jasmine-standalone-2.5.2/spec/invertedIndex.spec.js

```diff
@@ -0,0 +1,211 @@
+    });
+
+    it('should return true if file has property "title" and "text" ', () => {
+      expect(InvertedIndex.validateFile(book)).toBeTruthy();
+    });
+
+    it('should return false if file does not have property "title" and "text"',
+     () => {
+       expect(InvertedIndex.validateFile(invalidTitleAndText)).toBeFalsy();
+     });
+
+    it('should return false if file is not an array of JSON object',
+     () => {
+       expect(InvertedIndex.validateFile(notArray)).toBeFalsy();
+     });
+
+    it('should return false if file is an empty array',
+     () => {
+       expect(InvertedIndex.validateFile(emptyArrayFile)).toBeFalsy();
+     });
+
+    it('should return false if file is a JSON but not an array of an array',
+     () => {
+       expect(InvertedIndex.validateFile(invalid)).toBeFalsy();
+     });
+  });
+
+  describe('Read file', () => {
+    it('should be defined', () => {
+      expect(InvertedIndex.readFile).not.toBeUndefined();
+    });
+  });
+
+  describe('Tokenize words', () => {
+    it('should check that tokens are split and in sorted order', () => {
+      let test =
+        {
+          title: 'Alice in Wonderland',
+          text: 'Falls into a hole.'
+        };
+      const tokens =
+        ['a', 'alice', 'falls', 'hole', 'in', 'into', 'wonderland'];
+      test = InvertedIndex.tokenize(test);
+      expect(tokens).toEqual(test);
+    });
+  });
+
+  describe('createIndex', () => {
+    it('should return false if index is not created', () => {
+      expect(invertedIndex.createIndex(invalid)).toEqual(false);
+    });
+
+    it('should return true if index is created', () => {
+      expect(invertedIndex.createIndex('books.json', book)).toEqual(true);
+    });
+  });
+
+  describe('getIndices', () => {
+    it('should verify that index has been created', () => {
+      expect(Object.keys(invertedIndex.getIndices('books.json')).length)
+      .toBeGreaterThan(0);
+    });
+
+    it('should check that index maps the string to the correct objects in json'
+     + ' array', () => {
+      const createdIndex = {
+        a: [0],
+        alice: [0],
+        and: [0],
+        enters: [0],
+        falls: [0],
+        full: [0],
+        hole: [0],
+        imagination: [0],
+        in: [0],
+        into: [0],
+        of: [0],
+        rabbit: [0],
+        wonderland: [0],
+        world: [0]
+      };
+      let result = {};
+      result = invertedIndex.getIndices('anotherBook.json');
+      expect(Object.keys(result)).toEqual(Object.keys(createdIndex));
+      expect(values(result)).toEqual(values(createdIndex));
+    });
+  });
+
+  describe('searchIndices', () => {
+    it('should return true if search term is a string', () => {
+      const words = 'I love Barbie and Alice';
+      expect(Object.keys(invertedIndex.searchIndices(words,
+        'book.json'))).toBeTruthy();
+    });
+
+    it('should return true if search term is a string', () => {
```

Comment:
Thanks.
My implementation covers for both strings and arrays. There is a test case for array in my test file.